### PR TITLE
Support ServeError action and http_custom_errors phase

### PR DESCRIPTION
**Pull request by kroosec**
Add support for the new http_custom_errors phase, that uses the serve_error action. This action supports action parameters content, content_type and status_code.

### CODE REVIEW

File: rulesets.go [hunk 1]
```diff
@@ -84,6 +85,7 @@ func RulesetPhaseValues() []string {
 	return []string{
 		string(RulesetPhaseDDoSL4),
 		string(RulesetPhaseDDoSL7),
+		string(RulesetPhaseHTTPCustomErrors),
```
[the remark this answers is not in the document]
Comment (kroosec) [reply rulesets.go:89]: Indeed, thanks. Updated it.

**Review by jacobbednarz — APPROVED** [review]
LGTM. just pending public docs and we can merge this one.

File: .changelog/998.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+rulesets: add http_custom_errors phase and serve_error action
+```
+
```
Comment (jacobbednarz) [inline .changelog/998.txt]:
~~~suggestion
```release-note:enhancement
rulesets: add support for `http_custom_errors` phase
```

```release-note:enhancement
rulesets: add support for `serve_error` action
```
~~~~

**Comment by jacobbednarz** [comment]
public docs will be released early next week.

**Pull request merged**